File: ledger/ledger-common/src/main/java/net/corda/v5/ledger/common/transaction/CordaPackageSummary.java
package net.corda.v5.ledger.common.transaction;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

/**
 * Defines the summary of the metadata of a Corda package (CPK or CPI).
 */
public interface CordaPackageSummary {

    /**
     * Gets the name of the package.
     *
     * @return Returns the name of the package.
     */
    @NotNull
    String getName();

    /**
     * Gets the version of the package.
     *
     * @return Returns the version of the package.
     */
    @NotNull
    String getVersion();

    /**
     * Gets the hash sum identifying the signer for CPIs, or null for CPKs.
     *
     * @return Returns the hash sum identifying the signer for CPIs, or null for CPKs.
     */
    @Nullable
    String getSignerSummaryHash();

    /**
     * Gets the checksum of the package file.
     *
     * @return Returns the checksum of the package file.
     */
    @NotNull
    String getFileChecksum();
}

File: ledger/ledger-common/src/main/kotlin/net/corda/v5/ledger/common/transaction/CordaPackageSummary.kt
package net.corda.v5.ledger.common.transaction

/**
 * Summary of the metadata of a corda package (CPK or CPI)
 */
interface CordaPackageSummary {
    /**
     * @property name Name of the package
     */
    val name: String

    /**
     * @property version Version of the pacakge
     */
    val version: String

    /**
     * @property signerSummaryHash hash sum identifying the signer for signed packages (CPIs). Null for CPKs
     */
    val signerSummaryHash: String?

    /**
     * @property fileChecksum checksum of the package file
     */
    val fileChecksum: String
}